packages/zudoku/src/app/entry.server.tsx

```diff
@@ -8,31 +8,31 @@ import {
   createStaticHandler,
   createStaticRouter,
   isRouteErrorResponse,
+  type RouteObject,
 } from "react-router";
 import "virtual:zudoku-theme.css";
 import "vite/modulepreload-polyfill";
 import { BootstrapStatic, ServerError } from "zudoku/components";
-import type { ZudokuConfig } from "../config/config.js";
 import type { FileWritingResponse } from "../vite/prerender/FileWritingResponse.js";
 import "./main.css";
 import { getRoutesByConfig } from "./main.js";
-
 export { getRoutesByConfig };
 
 export const render = async ({
   template,
   request: baseRequest,
   response,
-  config,
+  routes,
+  basePath,
 }: {
   template: string;
   request: express.Request | Request;
   response: express.Response | FileWritingResponse;
-  config: ZudokuConfig;
+  routes: RouteObject[];
+  basePath?: string;
 }) => {
-  const routes = getRoutesByConfig(config);
   const { query, dataRoutes } = createStaticHandler(routes, {
```

Comment:
Passing`routes` in from the outside instead of running `getRoutesByConfig` is by far the most crucial performance gain